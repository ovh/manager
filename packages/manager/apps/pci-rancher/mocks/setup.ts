import { setupWorker } from 'msw/browser';
import { toMswHandlers } from './utils';
import { getRancherMocks } from './handlers';

export const setupMocks = async () =>
  setupWorker(...toMswHandlers(getRancherMocks())).start({
    onUnhandledRequest: 'bypass',
  });
