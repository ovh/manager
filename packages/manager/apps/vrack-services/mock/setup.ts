import { setupWorker } from 'msw/browser';
import { toMswHandlers } from '../../../../../playwright-helpers/msw';
import { getConfig } from './handlers';

export const setupMocks = async () =>
  setupWorker(
    ...toMswHandlers(
      getConfig({
        nbVs: 19,
        delayedOrders: true,
        deliveringVrackServicesOrders: true,
        deliveringVrackOrders: false,
        isAuthMocked: true,
      }),
    ),
  ).start({
    onUnhandledRequest: 'bypass',
  });
