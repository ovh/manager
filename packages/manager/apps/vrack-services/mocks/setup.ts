import { setupWorker } from 'msw/browser';
import { toMswHandlers } from '../../../../../playwright-helpers/msw';
import { getConfig } from './handlers';

export const setupMocks = async () =>
  setupWorker(
    ...toMswHandlers(
      getConfig({
        nbVs: 19,
        delayedOrders: true,
        deliveringVrackServicesOrders: false,
        deliveringVrackOrders: false,
        updateServicesKo: true,
      }),
    ),
  ).start({
    onUnhandledRequest: 'bypass',
  });
