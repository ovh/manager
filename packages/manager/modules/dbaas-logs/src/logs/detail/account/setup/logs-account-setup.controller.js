export default class LogsAccountSetupCtrl {
  /* @ngInject */
  constructor(
    $q,
    $state,
    $stateParams,
    CucControllerHelper,
    CucCloudMessage,
    LogsAccountService,
    LogsHomeService,
    LogsDetailService,
  ) {
    this.$q = $q;
    this.$state = $state;
    this.$stateParams = $stateParams;
    this.serviceName = this.$stateParams.serviceName;
    this.CucControllerHelper = CucControllerHelper;
    this.CucCloudMessage = CucCloudMessage;
    this.LogsAccountService = LogsAccountService;
    this.LogsHomeService = LogsHomeService;
    this.LogsDetailService = LogsDetailService;
    this.passwordRules = this.LogsAccountService.getPasswordRules(false);

    this.initLoaders();
  }

  initLoaders() {
    this.service = this.CucControllerHelper.request
      .getHashLoader({
        loaderFunction: () =>
          this.LogsDetailService.getServiceDetails(this.serviceName).then(
            (service) => {
              this.userName = service.username;
              return service;
            },
          ),
      })
      .load();
    this.accountDetails = this.CucControllerHelper.request
      .getHashLoader({
        loaderFunction: () =>
          this.LogsHomeService.getAccountDetails(this.serviceName).then(
            (account) => {
              this.fullName = `${account.me.firstname} ${account.me.name}`;
              return account;
            },
          ),
      })
      .load();
  }

  resetPasswordRules() {
    this.passwordRules = this.LogsAccountService.getPasswordRules(true);
  }

  changePassword() {
    if (this.form.$invalid) {
      return this.$q.when();
    }
    this.CucCloudMessage.flushChildMessage();
    return this.LogsAccountService.changePassword(
      this.serviceName,
      this.newPassword,
      true,
    ).then(() => {
      this.$state.go(
        'dbaas-logs.detail.home',
        { serviceName: this.serviceName },
        { reload: true },
      );
    });
  }
}
