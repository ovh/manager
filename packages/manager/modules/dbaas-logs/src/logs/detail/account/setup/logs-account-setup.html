<div class="logs-change-password">
    <span
        class="oui-heading_3"
        data-translate="logs_password_get_started_title"
    ></span>
    <div class="mb-3">
        <span
            data-translate="logs_password_setup_message"
            data-translate-values="{userFullName : ctrl.fullName, userName: ctrl.userName}"
        ></span>
    </div>
    <form novalidate name="ctrl.form">
        <div
            class="oui-field"
            data-ng-class="{'oui-field_error': ctrl.form.$submitted && ctrl.form.new_password.$invalid}"
        >
            <label
                for="new_password"
                class="oui-label"
                data-translate="logs_password_setup_label"
            ></label>
            <div class="oui-field__content">
                <div class="oui-field-control oui-field-control_5">
                    <oui-password
                        id="new_password"
                        name="new_password"
                        model="ctrl.newPassword"
                        on-change="ctrl.validatePassword()"
                        required
                    >
                        <oui-password-rule
                            data-ng-repeat="rule in ctrl.passwordRules track by rule.message"
                            validator="rule.validator(modelValue)"
                        >
                            <span
                                data-translate="{{:: rule.message }}"
                                data-translate-values="rule.translateParams"
                            ></span>
                        </oui-password-rule>
                    </oui-password>
                </div>
            </div>
        </div>
        <div class="cui-form-actions">
            <oui-spinner data-ng-if="ctrl.saving.loading"></oui-spinner>
            <div data-ng-if="!ctrl.saving.loading">
                <button
                    type="submit"
                    class="oui-button oui-button_primary"
                    data-translate="logs_save"
                    data-ng-click="ctrl.changePassword()"
                ></button>
            </div>
        </div>
    </form>
</div>
