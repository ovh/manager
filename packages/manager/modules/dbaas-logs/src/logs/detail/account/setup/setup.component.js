import controller from './logs-account-setup.controller';
import template from './logs-account-setup.html';

export default {
  controller,
  controllerAs: 'ctrl',
  template,
};
