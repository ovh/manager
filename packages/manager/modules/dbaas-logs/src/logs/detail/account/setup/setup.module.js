import angular from 'angular';

import '@ovh-ux/manager-core';
import '@ovh-ux/ng-ovh-cloud-universe-components';
import '@uirouter/angularjs';
import 'angular-translate';
import 'ovh-api-services';
import '@ovh-ux/ui-kit';

import component from './setup.component';
import routing from './setup.routing';

const moduleName = 'ovhManagerDbaasLogsDetailAccountSetup';

angular
  .module(moduleName, [
    'ngOvhCloudUniverseComponents',
    'oui',
    'ovhManagerCore',
    'ovh-api-services',
    'pascalprecht.translate',
    'ui.router',
  ])
  .config(routing)
  .component('dbaasLogsDetailAccountSetup', component)
  .run(/* @ngTranslationsInject:json ./translations */);

export default moduleName;
