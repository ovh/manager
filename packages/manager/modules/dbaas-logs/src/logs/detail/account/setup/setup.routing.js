export default /* @ngInject */ ($stateProvider) => {
  $stateProvider.state('dbaas-logs.detail.setup', {
    url: '/setup',
    views: {
      logsContent: 'dbaasLogsDetailAccountSetup',
    },
    resolve: {
      breadcrumb: /* @ngInject */ ($translate) =>
        $translate.instant('dbaas_logs_setup'),
    },
  });
};
