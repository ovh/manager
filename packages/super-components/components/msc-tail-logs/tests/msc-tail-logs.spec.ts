import { setupSpecTest } from './setup';

describe('specs:msc-tail-logs', () => {
  it('renders without error', async () => {
    const { code } = await setupSpecTest();

    expect(code?.shadowRoot?.innerHTML).toMatchSnapshot();
  });

  it('renders colors according to the log type', async () => {
    const { code } = await setupSpecTest({
      source:
        'cloud/project/33333333333333333333/database/mongodb/33333333-33333-333333-3333333333/logs',
    });

    expect(code?.shadowRoot?.innerHTML).toMatchSnapshot();
  });

  it('renders a message if there is no log to display', async () => {
    const { codeBlock, tradFR } = await setupSpecTest({
      source:
        'cloud/project/emptylogs/database/mongodb/44444444444-4444-44444444444444/logs',
    });

    expect(codeBlock?.textContent).toEqual(tradFR.empty_logs);
  });

  describe('error handling', () => {
    it('displays an error if the source does not respond', async () => {
      const { codeBlock, tradFR } = await setupSpecTest({
        source: '/notfound',
      });

      expect(codeBlock?.textContent).toEqual(tradFR.service_error_message);
    });

    it('displays an error if the source answers with an error', async () => {
      const { codeBlock, tradFR } = await setupSpecTest({
        source:
          'cloud/project/servererror/database/mongodb/55555555555-5555-55555555555555/logs',
      });

      expect(codeBlock?.textContent).toEqual(tradFR.service_error_message);
    });
  });

  describe('auto-refresh', () => {
    it('does not if refresh by default', async () => {
      const { page, mock } = await setupSpecTest();

      expect(mock.history.get).toHaveLength(1);
      mock.resetHistory();

      await new Promise<void>((resolve) => {
        setTimeout(() => resolve(), 1);
      });
      await page.waitForChanges();

      expect(mock.history.get).toHaveLength(0);
    });

    it('refresh the content if refresh is activated', async () => {
      const { page, mock } = await setupSpecTest({
        autoRefresh: true,
        refreshInterval: 1,
      });

      expect(mock.history.get).toHaveLength(1);
      mock.resetHistory();

      await new Promise<void>((resolve) => {
        setTimeout(() => resolve(), 1);
      });
      await page.waitForChanges();

      expect(mock.history.get.length).toBeGreaterThanOrEqual(1);
    });
  });
});
