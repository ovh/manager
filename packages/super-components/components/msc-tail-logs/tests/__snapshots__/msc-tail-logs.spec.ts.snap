// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`specs:msc-tail-logs renders colors according to the log type 1`] = `"<table class=\\"msc-tail-logs-table-wrapper\\"><thead><tr><th>Timestamp</th><th>hostname</th><th>message</th></tr></thead><tbody><tr><td><osds-text level=\\"caption\\" size=\\"200\\" color=\\"warning\\">2023-08-07T13:44:43.000Z</osds-text></td><td><osds-text level=\\"caption\\" size=\\"200\\" color=\\"warning\\">your database server 1</osds-text></td><td><osds-text level=\\"caption\\" size=\\"200\\" color=\\"warning\\">WARNING - ACCESS - Client has attempted to reauthenticate as a single user {\\"user\\":{\\"db\\":\\"local\\",\\"user\\":\\"__system\\"}}</osds-text></td></tr><tr><td><osds-text level=\\"caption\\" size=\\"200\\" color=\\"error\\">2023-08-07T13:44:44.000Z</osds-text></td><td><osds-text level=\\"caption\\" size=\\"200\\" color=\\"error\\">your database server 1</osds-text></td><td><osds-text level=\\"caption\\" size=\\"200\\" color=\\"error\\">CRIT - You are disconnected on your database mongoDB</osds-text></td></tr><tr><td><osds-text level=\\"caption\\" size=\\"200\\" color=\\"default\\">2023-08-07T13:44:45.000Z</osds-text></td><td><osds-text level=\\"caption\\" size=\\"200\\" color=\\"default\\">your database server 1</osds-text></td><td><osds-text level=\\"caption\\" size=\\"200\\" color=\\"default\\">DEBUG - You are connected on your database mongoDB</osds-text></td></tr><tr><td><osds-text level=\\"caption\\" size=\\"200\\" color=\\"info\\">2023-08-07T13:44:46.000Z</osds-text></td><td><osds-text level=\\"caption\\" size=\\"200\\" color=\\"info\\">your database server 1</osds-text></td><td><osds-text level=\\"caption\\" size=\\"200\\" color=\\"info\\">INFO - You are connected on your database mongoDB</osds-text></td></tr><tr><td><osds-text level=\\"caption\\" size=\\"200\\" color=\\"info\\">2023-08-07T13:44:47.000Z</osds-text></td><td><osds-text level=\\"caption\\" size=\\"200\\" color=\\"info\\">your database server 1</osds-text></td><td><osds-text level=\\"caption\\" size=\\"200\\" color=\\"info\\">INFO - You are connected on your database mongoDB</osds-text></td></tr><tr><td><osds-text level=\\"caption\\" size=\\"200\\" color=\\"info\\">2023-08-07T13:44:48.000Z</osds-text></td><td><osds-text level=\\"caption\\" size=\\"200\\" color=\\"info\\">your database server 1</osds-text></td><td><osds-text level=\\"caption\\" size=\\"200\\" color=\\"info\\">INFO - You are connected on your database mongoDB</osds-text></td></tr></tbody></table>"`;

exports[`specs:msc-tail-logs renders without error 1`] = `"<table class=\\"msc-tail-logs-table-wrapper\\"><thead><tr><th>Timestamp</th><th>hostname</th><th>message</th></tr></thead><tbody><tr><td><osds-text level=\\"caption\\" size=\\"200\\" color=\\"info\\">2023-08-07T13:44:49.000Z</osds-text></td><td><osds-text level=\\"caption\\" size=\\"200\\" color=\\"info\\">your database server 1</osds-text></td><td><osds-text level=\\"caption\\" size=\\"200\\" color=\\"info\\">INFO - You are connected on your database mongoDB</osds-text></td></tr><tr><td><osds-text level=\\"caption\\" size=\\"200\\" color=\\"info\\">2023-08-07T13:44:48.000Z</osds-text></td><td><osds-text level=\\"caption\\" size=\\"200\\" color=\\"info\\">your database server 1</osds-text></td><td><osds-text level=\\"caption\\" size=\\"200\\" color=\\"info\\">INFO - You are connected on your database mongoDB</osds-text></td></tr><tr><td><osds-text level=\\"caption\\" size=\\"200\\" color=\\"info\\">2023-08-07T13:44:47.000Z</osds-text></td><td><osds-text level=\\"caption\\" size=\\"200\\" color=\\"info\\">your database server 1</osds-text></td><td><osds-text level=\\"caption\\" size=\\"200\\" color=\\"info\\">INFO - You are connected on your database mongoDB</osds-text></td></tr><tr><td><osds-text level=\\"caption\\" size=\\"200\\" color=\\"info\\">2023-08-07T13:44:46.000Z</osds-text></td><td><osds-text level=\\"caption\\" size=\\"200\\" color=\\"info\\">your database server 1</osds-text></td><td><osds-text level=\\"caption\\" size=\\"200\\" color=\\"info\\">INFO - You are connected on your database mongoDB</osds-text></td></tr><tr><td><osds-text level=\\"caption\\" size=\\"200\\" color=\\"info\\">2023-08-07T13:44:45.000Z</osds-text></td><td><osds-text level=\\"caption\\" size=\\"200\\" color=\\"info\\">your database server 1</osds-text></td><td><osds-text level=\\"caption\\" size=\\"200\\" color=\\"info\\">INFO - You are connected on your database mongoDB</osds-text></td></tr><tr><td><osds-text level=\\"caption\\" size=\\"200\\" color=\\"info\\">2023-08-07T13:44:44.000Z</osds-text></td><td><osds-text level=\\"caption\\" size=\\"200\\" color=\\"info\\">your database server 1</osds-text></td><td><osds-text level=\\"caption\\" size=\\"200\\" color=\\"info\\">INFO - You are connected on your database mongoDB</osds-text></td></tr></tbody></table>"`;
